.github/workflows/ci.yml

```diff
@@ -35,28 +38,40 @@ jobs:
       matrix:
         os: [ubuntu-latest]
         rust: [stable, 1.45.0, beta]
+        args: ["", "--all-features"] 
     steps:
     - name: Checkout repository
       uses: actions/checkout@v1
     - name: Install Rust
       uses: hecrj/setup-rust-action@v1
       with:
         rust-version: ${{ matrix.rust }}
-    - run: cargo build --verbose
-    - run: cargo test --verbose 2>&1
-  allfeatures:
-    name: allfeatures
+    - run: cargo build --verbose ${{ matrix.args }}
+    - run: cargo test --verbose ${{ matrix.args }} 2>&1
+
+  coverage:
```

Comment:
One issue with this approach is that while it uses a separate job to the main `build` CI job, it is still part of the CI workflow which means that:

- the CI workflow will fail if generating the code coverage report fails
- the CI workflow will take longer to complete than it does without code coverage

Perhaps it would be good to investigate if it can be separated out to a separate workflow which is triggered after CI workflow success?

Reply:
Done, but the new coverage workflow is not visible in the Actions tab.

I suspect that GH won’t create the new coverage workflow until the new YML file is merged to master. As such we have no way of knowing if the new coverage YML is correct or is invoked by GH when intended, not will coverage reports be generated... 

@timbru: Unless you object I will add the new coverage.yml workflow file directly to master to see if it gets triggered and behaves as expected.

Reply:
The new coverage workflow is now working. I have tested that it runs on the correct Git commit when triggered by any of the following:
- A successful CI workflow run for a push to master (also if the CI workflow is re-run). See [this example workflow run](https://github.com/NLnetLabs/krill/actions/runs/486345094)
- A successful CI workflow run for a push to a pull request branch (also if the CI workflow is re-run). See [this example workflow run](https://github.com/NLnetLabs/krill/actions/runs/486372881)

A couple of additional PROs and CONs of this approach occurred to me:
- PRO: the coverage report workflow job doesn't add noise to the "checks" report shown in the GitHub view of a PR.
- CON: if we later wanted to fail builds because of insufficient or regressing coverage this approach would not work. However, with the current tooling and implementation the coverage report takes so long to generate that I don't think we'll want to make normal CI builds wait for a coverage report, and we could in theory (with more work), if we wanted to, make the separate coverage workflow post a comment on a PR about coverage issues.

This branch also might have solved the flakey test issue reported in #377, at least I pushed a couple of tweaks for #377 to this branch (because #377 was preventing the CI or coverage jobs to fail which blocked my work on this issue) and I haven't encounted the failure since.